monkey/infection_monkey/network/relay/relay_connection_handler.py

```diff
@@ -34,7 +34,18 @@ def handle_new_connection(self, sock: socket.socket):
             self._relay_user_handler.disconnect_user(addr)
         else:
             try:
-                self._pipe_spawner.spawn_pipe(sock)
+                self._pipe_spawner.spawn_pipe(sock, self._handle_pipe_data)
                 self._relay_user_handler.add_relay_user(addr)
```

Comment:
I think these two statements need to swap places. There's a race where `_handle_pipe_data()` is called before `add_relay_user()`, which would result in `renew_relay_user_membership()` being called before the membership exists.

I realize there's an `if` check in `renew_relay_user_membership()` to guard against this, but if that changes later on a bug could be introduced.

Reply:
Nice catch!